utils/service_preferences.ts

```diff
@@ -88,3 +97,36 @@ export function getServicePreferenceSettingsVersion(
     TE.mapLeft(_ => Error("Service Preferences Version < 0 not allowed"))
   );
 }
+
+export type ServicePreferencesForSpecialServices = (params: {
+  readonly serviceId: ServiceId;
+  readonly fiscalCode: FiscalCode;
+  readonly servicePreferences: ServicePreference;
+}) => TE.TaskEither<IResponseErrorQuery, ServicePreference>;
+
+export const getServicePreferencesForSpecialServices = (
+  activationModel: ActivationModel
+): ServicePreferencesForSpecialServices => ({
+  serviceId,
+  fiscalCode,
+  servicePreferences
+}): TE.TaskEither<IResponseErrorQuery, ServicePreference> =>
+  pipe(
+    activationModel.findLastVersionByModelId([serviceId, fiscalCode]),
+    TE.mapLeft(err =>
+      ResponseErrorQuery("Error reading service Activation", err)
+    ),
+    TE.map(_ => {
+      if (O.isNone(_) || _.value.status !== ActivationStatusEnum.ACTIVE) {
+        // When the Activation is missing the default value is INACTIVE.
+        return {
+          ...servicePreferences,
+          is_inbox_enabled: false
+        };
+      }
+      return {
+        ...servicePreferences,
+        is_inbox_enabled: true
+      };
+    })
```

Comment:
```suggestion
        TE.map(activation =>
      pipe(
        activation,
        O.filter(v => v.status === ActivationStatusEnum.ACTIVE),
        O.foldW(
          () => ({ ...servicePreferences, is_inbox_enabled: false }),
          _ => ({
            ...servicePreferences,
            is_inbox_enabled: true
          })
        )
      )
    )
```
We could make the code less imperative with something like this.

Reply:
very good suggestion @fabriziopapi !

I guess you can make it terser if instead of `Te.map(activation => pipe(activation, ...)` you just use `Te.map(flow(O.filter ...`